### PR TITLE
Add provisioning method support

### PR DESCRIPTION
**Pull request by nilasini**
## Purpose
related to https://github.com/wso2/product-is/issues/8047

Merging order
https://github.com/wso2/carbon-kernel/pull/2677
https://github.com/wso2/carbon-multitenancy/pull/191
https://github.com/wso2/identity-api-server/pull/157

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -524,8 +524,8 @@
         <identity.inbound.saml2.version>5.7.6</identity.inbound.saml2.version>
         <commons.beanutils.version>1.9.4</commons.beanutils.version>
         <mavan.findbugsplugin.exclude.file>findbugs-exclude-filter.xml</mavan.findbugsplugin.exclude.file>
-        <carbon.kernel.version>4.6.1-m3</carbon.kernel.version>
-        <carbon.multitenancy.version>4.8.4</carbon.multitenancy.version>
+        <carbon.kernel.version>4.6.1-m4</carbon.kernel.version>
```
Comment (nilasini) [inline pom.xml:527]: wait until kernel and multi-tenancy release

File: components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/java/org/wso2/carbon/identity/api/server/tenant/management/v1/core/ServerTenantManagementService.java [hunk 2]
```diff
@@ -196,6 +197,24 @@ private Tenant createTenantInfoBean(TenantModel tenantModel) throws TenantManage
             tenant.setAdminFirstName(tenantModel.getOwners().get(0).getFirstname());
             tenant.setAdminLastName(tenantModel.getOwners().get(0).getLastname());
             tenant.setEmail(tenantModel.getOwners().get(0).getEmail());
+            String provisioningMethod = tenantModel.getOwners().get(0).getProvisioningMethod();
+            if (INLINE.equalsIgnoreCase(provisioningMethod) || "invite-via-email".
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/java/org/wso2/carbon/identity/api/server/tenant/management/v1/core/ServerTenantManagementService.java:201]: extract to a constant

File: components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/java/org/wso2/carbon/identity/api/server/tenant/management/v1/core/ServerTenantManagementService.java [hunk 3]
```diff
@@ -196,6 +197,24 @@ private Tenant createTenantInfoBean(TenantModel tenantModel) throws TenantManage
...
+                        throw new TenantManagementClientException(TenantConstants.ErrorMessage.
+                                ERROR_CODE_MISSING_REQUIRED_PARAMETER.getCode(),
+                                String.format(TenantConstants.ErrorMessage.
+                                        ERROR_CODE_MISSING_REQUIRED_PARAMETER.getMessage(), "password"));
+                    }
+                    tenant.setAdminPassword(password);
+                }
+                tenant.setProvisioningMethod(provisioningMethod);
+            } else {
+                throw new TenantManagementClientException(TenantConstants.ErrorMessage.
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/java/org/wso2/carbon/identity/api/server/tenant/management/v1/core/ServerTenantManagementService.java:215]: provisioningMethod should be extensible. We better not limit to INLINE and invite-via-email. We don't need to check this and throw an error. we can write customizations based on the parameter value.

File: components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/resources/tenant-management.yaml [hunk 4]
```diff
@@ -451,6 +457,10 @@ components:
           type: string
           example: kim
           description: Last name of the owner.
+        provisioningMethod:
+          type: string
+          example: invite-via-email
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/resources/tenant-management.yaml:462]: lets put the sample as inline

**Pull request merged**